doc/UltiSnips.txt

```diff
@@ -55,9 +55,12 @@ UltiSnips                                      *snippet* *snippets* *UltiSnips*
 5. UltiSnips and Other Plugins                  |UltiSnips-other-plugins|
    5.1 Existing Integrations                    |UltiSnips-integrations|
    5.2 Extending UltiSnips                      |UltiSnips-extending|
-6. FAQ                                          |UltiSnips-FAQ|
-7. Helping Out                                  |UltiSnips-helping|
-8. Contributors                                 |UltiSnips-contributors|
+6. Debugging                                    |UltiSnips-Debugging|
```

Comment:
This is a great writeup, but I feel it is out of scope for the end user documentation. Let's move this into a separate document or the contributing document.

Reply:
I had the same though as you first, but then, I though that it could actually be useful when authoring snippet. Also, since there are user configurable options, having them into the doc permits to tag them (suppose you use it in your .vimrc to permanently set the host and port, and later you ask yourself what these variables are. With it in the doc, you can simple type `:help g:UltiSnips***`). If you still think it should not be in the user doc, then I can either move it to another help file, either put it in contributing

Reply:
I think it is a far shot to expect a user of UltiSnips to set up debugging through pdb. I consider myself a Python expert and only very very rarely used pdb before, so I think this is not a strong argument.

As for the tagging: That is understandable, but then again the variable names are descriptive enough that I would guess that they are not for the general public to use. I would search through the UltiSnips repo to figure out what they are. I hence still feel that this does not belong in this document, please move.

Reply:
That was an unexpected response since I use pdb in a daily base, and it was the first thing searched for in the documentation after I learned about python interpolation (without success hence this PR).

Is it ok if I create a new vim doc file named Ultisnips-advanced.txt ?

Reply:
> That was an unexpected response since I use pdb in a daily base, and it was the first thing I searched for in the documentation after I learned about python interpolation

One thing I learned in my 30+ years in computer science is that different people gravitate towards different tools. I have always been less productive in IDEs, but had collegues that could only work with them. I never had a lot of use for debuggers, thinking and reading through code usually helped me the most - you are clearly different there. I am therefore mostly enthusiastic about this change, because it is orthogonal to my workflow and hence might make it easier for people that do not work like me - this will make UltiSnips more accessible for other devs.

> Is it ok if I create a new vim doc file named Ultisnips-advanced.txt, and adding a small "debugging" paragraph pointing to this file ?

Yes, sure. I think it would be good to still reference it in the contributing doc though.

Reply:
Ok, I will change that tonight. It is already referenced in the contributing doc.

I like this debate. I agree with you, thinking an reading the code is the best. My use for the debugger is mainly to 1) stop right at the error and quickly inspect the variables to see if there is something wrong, and then read the code to know why, 2) When I start working on an existing code base, it's easier to understand the execution flow with breakpoints. I think the best is to try as much workflow as we can in our life so that we can select what works best for us and also understand how others thinks and work to ease communication :)

Reply:
Amen!